core/services/bulletprooftxmanager/eth_resender.go

```diff
@@ -0,0 +1,150 @@
+package bulletprooftxmanager
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/lib/pq"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/eth"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	"gorm.io/gorm"
+)
+
+// pollInterval is the maximum amount of time in addition to
+// EthTxResendAfterThreshold that we will wait before resending an attempt
+const defaultResenderPollInterval = 5 * time.Second
+
+// EthResender periodically picks up transactions that have been languishing
+// unconfirmed for a configured amount of time without being sent, and sends
+// their highest priced attempt again. This helps to defend against geth/parity
+// silently dropping txes, or txes being ejected from the mempool.
+//
+// Previously we relied on the bumper to do this for us implicitly but there
+// can occasionally be problems with this (e.g. abnormally long block times, or
+// if gas bumping is disabled)
+type EthResender struct {
+	db           *gorm.DB
+	ethClient    eth.Client
+	interval     time.Duration
+	ageThreshold time.Duration
+
+	chStop chan struct{}
+	chDone chan struct{}
+}
+
+func NewEthResender(db *gorm.DB, ethClient eth.Client, pollInterval, ethTxResendAfterThreshold time.Duration) *EthResender {
+	if ethTxResendAfterThreshold == 0 {
+		panic("EthResender requires a non-zero threshold")
+	}
+	return &EthResender{
+		db,
+		ethClient,
+		pollInterval,
+		ethTxResendAfterThreshold,
+		make(chan struct{}),
+		make(chan struct{}),
+	}
+}
+
+func (er *EthResender) Start() {
+	logger.Infof("EthResender: Enabled with poll interval of %s and age threshold of %s", er.interval, er.ageThreshold)
+	go er.runLoop()
```

Comment:
I think this is a good use case for `go gracefulpanic.WrapRecover`

Reply:
What will that give us?

Reply:
seems like scenario where we can degrade gracefully - if there's a panic in the EthResender we can error instead of crash and the node can still function